File: jsApps/launcher/src/service/userService/Mapper.ts
import AuthResponse from "./models/AuthModels/AuthResponse";
import User from "./models/User";

export const MapAuthResponseToUser = (response: AuthResponse): User => {
    return {
        hash: response.hash,
        jwt: response.jwt,
        name: response.name,
        roles: response.roles,
        surname: response.surname,
        username: response.surname,
        expireDate: response.expireDate,
        companyHash: response.companyHash,
    };
};

export default MapAuthResponseToUser;

File: Deliver/Models/Request/User/Mapper.cs
using Models.Response.Authentication;

namespace Models.Request.User
{
    public static class Mapper
    {
        public static Db.User AsUser(this CreateUserRequest createUserRequest)
            => new()
            {
                Name = createUserRequest.Name,
                Surname = createUserRequest.Surname,
                Email = createUserRequest.Email,
                PhoneNumber = createUserRequest.PhoneNumber,
                Username = createUserRequest.Username,
            };

        public static Db.User UpdateUser(this UpdateUserRequest updateUser, Db.User user)
        {
            user.Email = updateUser.Email;
            user.PhoneNumber = updateUser.PhoneNumber;
            user.Name = updateUser.Name;
            user.Surname = updateUser.Surname;
            return user;
        }
    }
}
